lazy_tensor_core/lazy_tensor_core/csrc/compiler/backend_impl_interface.h

```diff
@@ -25,38 +55,64 @@ class BackendImplInterface {
   virtual std::unique_ptr<ir::LoweringContext> CreateLoweringContext(
       const std::string& name, Device device) const = 0;
 
-  virtual lazy_tensors::ComputationClient* GetComputationClient() const = 0;
+  // TODO(whc) need to keep this?
+  virtual std::vector<std::string> GetCompilationDevices(
+      const std::string& device, c10::ArrayRef<std::string> devices) const = 0;
+
+  virtual std::vector<ComputationPtr> Compile(
+      std::vector<ComputationPtr> instances) const = 0;
 
-  virtual lazy_tensors::ComputationClient* GetComputationClientIfInitialized()
+  virtual std::vector<BackendDataPtr> ExecuteComputation(
+      Computation& computation, c10::ArrayRef<BackendDataPtr> arguments,
+      const std::string& device) const = 0;
+
+  /**
+   * Device Configuration
+   * */
+
+  // Identifies a 'host' in distributed setting? Used to hash a particular
+  // host's graph?
+  // -- why do we need this, and should it be renamed if we do?  Should it be
+  // one per gpu in mgpu setting?
+  virtual std::string GetResourceDomain(const std::string& device) const = 0;
```

Comment:
@JackCaoG The APIs in this section are the most unclear to me.  As in, we don't use them yet in TS backend, and we don't yet have a design for distributed.  I am wondering if we can delete most of them now, then add back relevant ones later as we do the design.

Do you know what the minimal set of device related APIs is for you to be able to use this interface for bootstrapping?  

Did you reach a conclusion about whether you need different 'compilation devices' APIs vs actual physical devices?

Reply:
I think it is OK to delete the unused functions and think more about the API when we do the design.

I have a feeling that having a separate `compilation` device might be a xla specified requirement. We can remove it from LTC for now but I am hoping that we can have a way PT/XLA to be able to extend class/methods like `Computation`, `Data` and `Compile` to inject some XLA specific logic like `Layout` and `Compilation Device`.